frontend/src/app/components/homepage/homepage.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, OnInit } from '@angular/core';
+import { RestService } from '../../shared/services/rest/rest.service';
+
+@Component({
+	selector: 'app-homepage',
+	templateUrl: './homepage.component.html',
+	styleUrls: ['./homepage.component.css']
+})
+export class HomepageComponent implements OnInit {
+
+	books: any[];
+	fakeBooks = [
+		{ 'id': 1, 'title': 'title 1', 'authorName': 'name1', 'authorSurname': 'surname1' },
+		{ 'id': 2, 'title': 'title 2', 'authorName': 'name2', 'authorSurname': 'surname2' },
+		{ 'id': 3, 'title': 'title 3', 'authorName': 'name2', 'authorSurname': 'surname2' }
+	];
+	newBook = { 'id': null, 'title': 'book ' + Math.floor(Math.random() * 100), 'authorName': 'name ' + Math.floor(Math.random() * 100), 'authorSurname': 'surname' + Math.floor(Math.random() * 100) };
+
+	constructor(private restService: RestService) {
+	}
+
+	ngOnInit() {
+
+		const url = 'library/books';
+		// get all books
+		this.restService.getAll('library/books').then(response => {
```

Comment:
Powinieneś stworzyć serwis pośredniczący w komunikacji z BE. Komponent nie powinien nawet wiedzieć czy te dane są pobierane przez api, czy z mocka, czy losowane. On ma je wyświetlać, tyle.
Zrób serwis BooksService, on zawsze zwróci listę książek i w przypadku błędu wyświetli komunikat dla użytkownika. Tutaj z 20 lini zrobią się 2 (pod warunkiem że zaczniesz korzystać z tego: http://www.syntaxsuccess.com/viewarticle/async-and-await-in-typescript